src/Plugin/Condition/DataListContains.php

```diff
@@ -39,6 +39,20 @@ class DataListContains extends RulesConditionBase {
   /**
    * {@inheritdoc}
    */
+  public function refineContextdefinitions() {
+
+    // Refine the item context
+    if($item = $this->getContextValue('item')) {
+      if ($item instanceof EntityInterface && $type = $item->getTypedData('type')) {
```

Comment:
->getTypeData() will return an object, but we want the entity type ID here, so it should be getEntityTypeId().